agent/uiserver/buffered_file.go

```diff
@@ -0,0 +1,69 @@
+package uiserver
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"time"
+)
+
+// bufferedFile implements os.File and allows us to modify a file from disk by
+// writing out the new version into a buffer and then serving file reads from
+// that.
+type bufferedFile struct {
+	buf  *bytes.Reader
+	info os.FileInfo
+}
+
+func newBufferedFile(buf []byte, info os.FileInfo) *bufferedFile {
+	return &bufferedFile{
+		buf:  bytes.NewReader(buf),
+		info: info,
+	}
+}
+
+func (t *bufferedFile) Read(p []byte) (n int, err error) {
+	return t.buf.Read(p)
+}
+
+func (t *bufferedFile) Seek(offset int64, whence int) (int64, error) {
+	return t.buf.Seek(offset, whence)
+}
+
+func (t *bufferedFile) Close() error {
+	// Reset the read cursor since the next "open" will assume it is reading from
```

Comment:
Did you mean to actually Seek(0,0) here?

Reply:
Heh I did when i wrote that comment, and then I made it work a different way be refactoring this to wrap a `bytes.Reader` instead of a `bytes.Buffer` so each "instance" had it's own cursor state. I've removed the comment in next commit! Good 👀 .